File: service/backend/src/main/java/ru/tykvin/hermes/file/storage/Storage.java
package ru.tykvin.hermes.file.storage;

import lombok.SneakyThrows;
import ru.tykvin.hermes.file.model.DownloadingEntity;
import ru.tykvin.hermes.file.model.FilesystemStorageFile;
import ru.tykvin.hermes.file.model.UploadingEntity;
import ru.tykvin.hermes.model.User;

import javax.servlet.http.HttpServletRequest;

import java.io.File;
import java.util.List;
import java.util.UUID;

public interface Storage {

    @SneakyThrows
    FilesystemStorageFile write(UploadingEntity uploadingEntity);
}

File: src/main/kotlin/ru/tykvin/hermes/storage/Storage.kt
package ru.tykvin.hermes.storage

import ru.tykvin.hermes.file.service.dao.FilesystemStorageFile
import ru.tykvin.hermes.file.service.dao.UploadingEntity

interface Storage {
    fun write(uploadingEntity: UploadingEntity): FilesystemStorageFile
}
